File: matlab/params/PARAMS_landau_damping.m
params = struct();
params.mycase = "landau_damping";          % "two_stream"
params.Nx = 2^8;                            % Number of spatial grid points
params.Nv = 2^8;                            % Number of velocity grid points
params.Ns = 1;                              % Number of species (electrons and ions)
params.method="predcorr";
params.species_name = ["electrons","ions"];  % name of the different species
params.Mr = 1;                 % Mass ratio for ions
params.Mass = [1, params.Mr];          % Mass of species
params.charge = [1, 1];          % Charge of species
params.Nt_max = 4000;             % Maximum number of time steps
params.dt = 1/2;                   % Time step size
params.Tend = 40;                % End time of simulation
params.dt_save = 10;                        % Save after dt_save time

% Initial condition parameters
params.k = 0.5;                   % Wave number
params.alpha = 0.2;               % Perturbation amplitude
params.pert = @(x) params.alpha * cos(params.k * x); % Perturbation function

% Electrons
params.Ue = 0;                   % Electron drift velocity
params.fe0 = @(x, v) (1+params.pert(x)) ./ (sqrt(2 * pi)) .* (exp(-(v).^2 / 2)); % Electron distribution
params.fi0 = @(x, v) sqrt(params.Mr / (2 * pi)) .* (exp(-params.Mr * (v).^2 / 2)); % Ion distribution
params.fini = {params.fe0,params.fi0};
% Grid parameters
params.Lx = 2 * pi / params.k;    % Spatial domain length
params.Lv_s = [12, 0.1 * pi]; % Velocity domain lengths for each species
params.Lv = 12;

File: params/PARAMS_landau_damping.py
import jax.numpy as jnp

### General parameters ###
mycase         = "landau_damping"          # "two_stream"
Nx             = 2**8                      # Number of spatial grid points
Nv             = 2**8                      # Number of velocity grid points
Ns             = 1                         # Number of species (electrons and ions)
method         = "predcorr"
species_name   = ["electrons", "ions"]     # Name of the different species
Mr             = 1                         # Mass ratio for ions
Mass           = [1, Mr]                   # Mass of species
charge         = [-1, 1]                   # Charge of species
Nt_max         = 4000                      # Maximum number of time steps
dt             = 1/2                       # Time step size

Tend           = 40                        # End time of simulation

### Initial condition parameters ###
k      = 0.5                               # Wave number
alpha  = 0.2                               # Perturbation amplitude
pert   = lambda x: alpha * jnp.cos(k * x)  # Perturbation function

### Electrons ###
Ue     = 0                                                                            # Electron drift velocity
fe0    = lambda x, v: (1 + pert(x)) / (jnp.sqrt(2 * jnp.pi)) * jnp.exp(- v ** 2 / 2)  # Electron distribution
fi0    = lambda    v: jnp.sqrt(Mr / (2 * jnp.pi)) * jnp.exp(-Mr * v ** 2 / 2)         # Ion distribution
fini   = [fe0, fi0]

### Grid parameters ###
Lx     = 2 * jnp.pi / k                    # Spatial domain length
Lv_s   = [12, 0.1 * jnp.pi]                # Velocity domain lengths for each species